apps/core/lib/core/snitch/data/model/payment/card_payment.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Core.Snitch.Data.Model.CardPayment do
+  @moduledoc """
+  CardPayment API and utilities.
+
+  `CardPayment` is a concrete payment subtype in Snitch. By `create/2`ing a
+  CardPayment, the supertype Payment is automatically created in the same
+  transaction.
+
+  > For other supported payment sources, see
+    `Core.Snitch.Data.Schema.PaymentMethod`
+  """
+  use Core.Snitch.Data.Model
+
+  @doc """
+  Creates both `Payment` and `CardPayment` records in a transaction for Order
+  represented by `order_id`.
+
+  * `payment_params` are validated using
+    `Core.Snitch.Data.Schema.Payment.changeset/3` with the `:create` action and
+    because `slug` and `order_id` are passed explicitly to this function,
+    they'll be ignored if present in `payment_params`.
+  * `card_params` are validated using
+  `Core.Snitch.Data.Schema.CardPayment.changeset/3` with the `:create` action.
+  """
+  @spec create(String.t(), non_neg_integer(), map, map) ::
+          {:ok, %{card_payment: Schema.CardPayment.t(), payment: Schema.Payment.t()}}
+          | {:error, Ecto.Changeset.t()}
+  def create(slug, order_id, payment_params, card_params) do
+    payment = struct(Schema.Payment, payment_params)
+    card_method = Model.PaymentMethod.get_card()
+
+    more_payment_params = %{
+      order_id: order_id,
+      payment_type: "ccd",
+      payment_method_id: card_method.id,
+      slug: slug
+    }
+
+    payment_changeset = Schema.Payment.changeset(payment, more_payment_params, :create)
+
+    Ecto.Multi.new()
+    |> Ecto.Multi.insert(:payment, payment_changeset)
+    |> Ecto.Multi.run(:card_payment, fn %{payment: payment} ->
+      all_card_params = Map.put(card_params, :payment_id, payment.id)
```

Comment:
I am not sure if it is already done before passing to this method,
but could we have had a `changeset` for  `card_params`/ `all_card_params`  before running `create` query  on `CardPayment`.
Any Thoughts on this ?
cc: @ashish173  @pkrawat1

Reply:
`QH.create(Schema.CardPayment, all_card_params, Repo)` (line 45) builds the changeset.